components/webgpu/canvas_context.rs

```diff
@@ -579,7 +579,7 @@ impl crate::WGPU {
                             staging_buffer
                                 .snapshot()
                                 .unwrap_or_else(|| Snapshot::cleared(configuration.size))
-                                .as_ipc(),
+                                .to_shared(),
```

Comment:
We can avoid vec alloc here too.

Reply:
I suspect the most common case is to create cleared for sending, so we might need to check all occurrences.

Reply:
I've fixed this and reviewed the rest of the callsites. It seems that the other are creating cleared Snapshots for either:

1. Resetting an internally stored `Snapshot` in a DOM object.
2. Returning an empty Snapshot for the purposes of things like `createImageBitmap`.

I didn't see any other callsites that trivially sent the immediately constructed `Snapshot`.